persistent-mysql/Database/Persist/MySQL.hs

```diff
@@ -827,9 +827,19 @@ data ColumnInfo = ColumnInfo
 -- @INFORMATION_SCHEMA@ tables.
 parseColumnType :: Text -> ColumnInfo -> ExceptT String IO (SqlType, Maybe Integer)
 -- Ints
-parseColumnType "tinyint" ci | ciColumnType ci == "tinyint(1)" = return (SqlBool, Nothing)
-parseColumnType "int" ci | ciColumnType ci == "int(11)"        = return (SqlInt32, Nothing)
-parseColumnType "bigint" ci | ciColumnType ci == "bigint(20)"  = return (SqlInt64, Nothing)
+-- The display width is deprecated and being removed in MySQL 8.X.  To be
+-- consistent with earlier versions, which do report it, accept either
+-- the bare type in `ciColumnType ci`, or the type adorned with the expected
+-- value for the display width (ie the defaults for int and bigint, or the
+-- value explicitly set in `showSqlType` for SqlBool).
+--
+parseColumnType "tinyint" ci
+    | ciColumnType ci == "tinyint" || ciColumnType ci == "tinyint(1)" = return (SqlBool, Nothing)
+parseColumnType "int" ci
+    | ciColumnType ci == "int"     || ciColumnType ci == "int(11)"    = return (SqlInt32, Nothing)
+parseColumnType "bigint" ci
+    | ciColumnType ci == "bigint"  || ciColumnType ci == "bigint(20)" = return (SqlInt64, Nothing)
```

Comment:
does it maybe make sense to do a `Just _ <- Text.stripPrefix "tinyint" ` pattern? That would match `tinyint`, `tinyint(1)`, and `tinyint(2345)` etc.

Reply:
Again, I am trying to avoid changing anything for MySQL 5.7 and early 8.X.  These versions will always report the display width, so I thought it best to match it exactly, as before, to avoid including cases which were not included in `persistent-mysql-2.13.1.0`.  The addition is the matching of the bare type, eg `int`, which will only be reported by later 8.X versions